processors/opentelemetry_baggage_processor/src/otel_baggage_processor.erl

```diff
@@ -0,0 +1,65 @@
+-module(otel_baggage_processor).
+
+-behaviour(otel_span_processor).
+
+-include_lib("opentelemetry/include/otel_span.hrl").
+-include_lib("opentelemetry_api/include/opentelemetry.hrl").
+
+-export([on_start/3, on_end/2, force_flush/1]).
+
+-type processor_config() :: term().
+
+-spec on_start(otel_ctx:t(), opentelemetry:span(), processor_config()) ->
+                opentelemetry:span().
+on_start(Ctx, Span, Config) ->
+  Baggage = otel_baggage:get_all(Ctx),
+  Attributes =
+    maps:fold(fun(Key, {Value, Metadata}, Attributes) ->
+                 NewKey = add_prefix(Key, Config),
```

Comment:
Can you take `Prefix` out of `Config` before calling `maps:fold`? That way it isn't getting the same value out of the map every call to `add_prefix` but only once.

Reply:
Done